File: em Vídeo/POO [60 Horas]/PHP POO [40 Horas]/Aula 08/Luta.php
<?php
    require_once '..//Aula 07//Lutador.php';
    class Luta {
        private $desafiado;
        private $desafiante;
        private $rounds;
        private $aprovada;

        function getDesafiado() {
            return $this->desafiado;
        }
    
        function setDesafiado($desafiado) {
            $this->desafiado = $desafiado;
        }
    
        function getDesafiante() {
            return $this->desafiante;
        }
    
        function setDesafiante($desafiante) {
            $this->desafiante = $desafiante;
        }
    
        function getRounds() {
            return $this->rounds;
        }
    
        function setRounds($rounds) {
            $this->rounds = $rounds;
        }
    
        function isAprovada() {
            return $this->aprovada;
        }
    
        function setAprovada($aprovada) {
            $this->aprovada = $aprovada;
        }
    
        function marcarLuta($l1, $l2) {
            if (($l1->getCategoria() == $l2->getCategoria())
                    && ($l1 != $l2)) {
                $this->setAprovada(true);
                $this->setDesafiado($l1);
                $this->setDesafiante($l2);
            } else {
                $this->setAprovada(false);
                $this->setDesafiado(null);
                $this->setDesafiante(null);
            }
        }
    
        function lutar() {
            if ($this->aprovada) {
                print("### DESAFIADO ###"."\n");
                $this->desafiado->apresentar();
                print("### DESAFIANTE ###"."\n");
                $this->desafiante->apresentar();
    
                print("==== RESULTADO DA LUTA ===="."\n");
                $vencedor = rand(0, 2);
                switch ($vencedor) {
                    case 0:
                        print("Empate"."\n");
                        $this->desafiado->empatarLuta();
                        $this->desafiante->empatarLuta();
                        break;
    
                    case 1:
                        print("Vitória do: ".$this->desafiado->getNome()."\n");
                        $this->desafiado->ganharLuta();
                        $this->desafiante->perderLuta();
                        break;
    
                    case 2:
                        print("Vitória do: ".$this->desafiante->getNome()."\n");
                        $this->desafiante->ganharLuta();
                        $this->desafiado->perderLuta();
                        break;
                }
                print("==========================="."\n");
            } else {
                print("A luta não pode acontecer"."\n");
            }
        }
    
    }

File: em Vídeo/POO [60 Horas]/Java POO [40 Horas]/Aula 08/Luta.java
import java.util.Random;

public class Luta {
    private Lutador desafiado;
    private Lutador desafiante;
    private int rounds;
    private boolean aprovada;

    public Lutador getDesafiado() {
        return desafiado;
    }

    public void setDesafiado(Lutador desafiado) {
        this.desafiado = desafiado;
    }

    public Lutador getDesafiante() {
        return desafiante;
    }

    public void setDesafiante(Lutador desafiante) {
        this.desafiante = desafiante;
    }

    public int getRounds() {
        return rounds;
    }

    public void setRounds(int rounds) {
        this.rounds = rounds;
    }

    public boolean isAprovada() {
        return aprovada;
    }

    public void setAprovada(boolean aprovada) {
        this.aprovada = aprovada;
    }

    public void marcarLuta(Lutador l1, Lutador l2) {
        if ((l1.getCategoria().equals(l2.getCategoria()))
                && (l1 != l2)) {
            this.setAprovada(true);
            this.setDesafiado(l1);
            this.setDesafiante(l2);
        } else {
            this.setAprovada(false);
            this.setDesafiado(null);
            this.setDesafiante(null);
        }
    }

    public void lutar() {
        if (this.aprovada) {
            System.out.println("### DESAFIADO ###");
            this.desafiado.apresentar();
            System.out.println("### DESAFIANTE ###");
            this.desafiante.apresentar();

            System.out.println("==== RESULTADO DA LUTA====");
            Random random = new Random();
            int vencedor = random.nextInt(3);
            switch (vencedor) {
                case 0:
                    System.out.println("Empate");
                    this.desafiado.empatarLuta();
                    this.desafiante.empatarLuta();
                    break;

                case 1:
                    System.out.println("Vitória do: " + this.desafiado.getNome());
                    this.desafiado.ganharLuta();
                    this.desafiante.perderLuta();
                    break;

                case 3:
                    System.out.println("Vitória do: " + this.desafiante.getNome());
                    this.desafiante.ganharLuta();
                    this.desafiado.perderLuta();
                    break;
            }
            System.out.println("==========================");
        } else {
            System.out.println("A luta não pode acontecer");
        }
    }

}
